src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/PasswordDeriveBytes.cs

```diff
@@ -0,0 +1,327 @@
+
+        public PasswordDeriveBytes(byte[] password, byte[] salt, string hashName, int iterations, CspParameters cspParams)
+        {
+            IterationCount = iterations;
+            Salt = salt;
+            HashName = hashName;
+            _password = password;
+            _cspParams = cspParams;
+        }
+
+        public string HashName
+        {
+            get { return _hashName; }
+            set
+            {
+                if (_baseValue != null)
+                    throw new CryptographicException(SR.Cryptography_PasswordDerivedBytes_ValuesFixed, nameof(HashName));
+
+                _hashName = value;
+                _hash = (HashAlgorithm)CryptoConfig.CreateFromName(_hashName);
+            }
+        }
+
+        public int IterationCount
+        {
+            get { return _iterations; }
+            set
+            {
+                if (value <= 0)
+                    throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_NeedPosNum);
+                if (_baseValue != null)
+                    throw new CryptographicException(SR.Cryptography_PasswordDerivedBytes_ValuesFixed, nameof(IterationCount));
+
+                _iterations = value;
+            }
+        }
+
+        public byte[] Salt
+        {
+            get
+            {
+                return (byte[])_salt?.Clone();
+            }
+            set
+            {
+                if (_baseValue != null)
+                    throw new CryptographicException(SR.Cryptography_PasswordDerivedBytes_ValuesFixed, nameof(Salt));
+
+                _salt = (byte[])value?.Clone();
+            }
+        }
+
+        [Obsolete("Rfc2898DeriveBytes replaces PasswordDeriveBytes for deriving key material from a password and is preferred in new applications.")]
+#pragma warning disable 0809 // obsolete member overrides non-obsolete member
+        public override byte[] GetBytes(int cb)
+        {
+            int ib = 0;
+            byte[] rgb;
+            byte[] rgbOut = new byte[cb];
+
+            if (_baseValue == null)
+            {
+                ComputeBaseValue();
+            }
+            else if (_extra != null)
+            {
+                ib = _extra.Length - _extraCount;
+                if (ib >= cb)
+                {
+                    Buffer.BlockCopy(_extra, _extraCount, rgbOut, 0, cb);
+
+                    if (ib > cb)
+                    {
+                        _extraCount += cb;
+                    }
+                    else
+                    {
+                        _extra = null;
+                    }
+
+                    return rgbOut;
+                }
+                else
+                {
+                    // Note: The second parameter should really be _extraCount instead.
+                    // However, changing this would constitute a breaking change.
+                    Buffer.BlockCopy(_extra, ib, rgbOut, 0, ib);
```

Comment:
The comment's a really good marker here, but if we don't have a test explicitly ensuring it doesn't change, can we add one?